lib/spack/llnl/util/filesystem.py

```diff
@@ -803,21 +800,27 @@ def copy_tree(
             ignore=ignore,
             follow_nonexisting=True,
         ):
-            if os.path.islink(s):
+            if islink(s):
                 link_target = resolve_link_target_relative_to_the_link(s)
                 if symlinks:
                     target = os.readlink(s)
-                    if os.path.isabs(target):
 
-                        def escaped_path(path):
-                            return path.replace("\\", r"\\")
+                    def escaped_path(path):
+                        return path.replace("\\", r"\\")
 
+                    if os.path.isabs(target):
                         new_target = re.sub(escaped_path(abs_src), escaped_path(abs_dest), target)
-                        if new_target != target:
-                            tty.debug("Redirecting link {0} to {1}".format(target, new_target))
-                            target = new_target
 
-                    symlink(target, d)
+                    else:
+                        new_target = re.sub(escaped_path(src), escaped_path(dest), target)
```

Comment:
This is necessary for moving the target of the link from `source/link/target` to `dest/link/target` so the symlink that is being copied over points to the `dest` link target and not the original `source` link target, regardless of whether the path is absolute or relative. Without this, the link points to the wrong file. The `re.sub` is a no-op if the relative path doesn't include the top-level "source" directory in the path. 

The `copy_tree` unit test was failing.

Reply:
> If I replace this line with new_target = target, then spack unit-test -k copytree (and spack unit-test -k filesystem.py) passes on Mac, so is this a Windows-specific failure? Could you add the error message here so I could understand it better.

Looking closer at the commit that changed this back in April, I think the issue is NOT that that symlinking fails, but rather that the new link (in `dest/`) pointed to the original file in `source/` e.g.

```bash
$ ls -l
dest/link_name -> source/target_name
```

whereas is should be

```bash
$ ls -l
dest/link_name -> dest/target_name
```

But like I said, if the link target was something like 
```bash
$ ls -l
dest/link_name -> ../target_name
```
then this is a no-op.

> While looking at this, I also noticed that the TestCopyTree and TestInstallTree classes are skipped on Windows: I think those should be enabled in this PR (unless they specific to Mac/Linux).

Enabled and passing with one modification the one test to use OS path separators instead of hardcoded `/` path sep.